command/build_test.go

```diff
@@ -1112,3 +1112,62 @@ func TestBuildCommand_ParseArgs(t *testing.T) {
 		})
 	}
 }
+
+// TestBuildCmd aims to test the build command, with output validation
+func TestBuildCmd(t *testing.T) {
+	tests := []struct {
+		name         string
+		args         []string
+		expectedCode int
+		outputCheck  func([]byte, []byte) error
+	}{
+		{
+			name: "hcl - no build block error",
+			args: []string{
+				testFixture("hcl", "no_build.pkr.hcl"),
+			},
+			expectedCode: 1,
+			outputCheck: func(out, err []byte) error {
+				strerr := string(err)
+				if !strings.Contains(strerr, "Error: Missing build block") {
+					return fmt.Errorf("expected 'Error: Missing build block' in output, did not find it")
+				}
+
+				nbErrs := strings.Count(strerr, "Error: ")
+				if nbErrs != 1 {
+					return fmt.Errorf("error: too many errors in stdout for build block")
+				}
+
+				return nil
+			},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			c := &BuildCommand{
+				Meta: TestMetaFile(t),
+			}
+
+			exitCode := c.Run(tt.args)
+			if exitCode != tt.expectedCode {
+				t.Errorf("process exit code mismatch: expected %d, got %d",
+					tt.expectedCode,
+					exitCode)
+			}
+
+			out, stderr := GetStdoutAndErrFromTestMeta(t, c.Meta)
+			outputValidateError := tt.outputCheck(out, stderr)
+			if outputValidateError != nil {
+				if len(out) != 0 {
+					t.Logf("command stdout: %q", string(out))
+				}
+
+				if len(stderr) != 0 {
+					t.Logf("command stderr: %q", string(stderr))
+				}
+				t.Error(outputValidateError.Error())
```

Comment:
We have a function within the existing tests that does exactly this except that it returns string values for stdout and stderr, which is fine for these tests. 

nit: you could rename `outputValidateError` to `err`. Since it is short lived and it is the return value of the call to tt.outputCheck(..) there is enough context to denote that this is an output check error. The name outputValidateError reads like a custom error type. 
```suggestion
			out, stderr := commandOutput(t, c.Meta)
			outputValidateError := tt.outputCheck(out, stderr)
			if outputValidateError != nil {
				if out != "" {
					t.Logf("command stdout: %q", out)
				}

				if stderr != "" {
					t.Logf("command stderr: %q", stderr)
				}
				t.Error(outputValidateError.Error())
```